catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java

```diff
@@ -108,35 +104,50 @@ public List<String> listTables(String databaseName) throws NoSuchSchemaException
   @Override
   public JdbcTable load(String databaseName, String tableName) throws NoSuchTableException {
     try (Connection connection = getConnection(databaseName)) {
-      String comment;
-      Map<String, String> properties;
-      try (ResultSet table = getTable(connection, tableName)) {
-        if (!table.next()) {
-          throw new NoSuchTableException("Table " + tableName + " does not exist.");
-        }
-        comment = table.getString("REMARKS");
-        properties = extractPropertiesFromResultSet(table);
+      // 1.Get table information
+      ResultSet table = getTable(connection, databaseName, tableName);
+      if (!table.next() || !tableName.equals(table.getString("TABLE_NAME"))) {
+        throw new NoSuchTableException(
+            String.format("Table %s does not exist in %s.", tableName, databaseName));
       }
-      JdbcColumn[] jdbcColumns;
-      try (ResultSet column = getColumns(connection, tableName)) {
-        List<JdbcColumn> result = new ArrayList<>();
-        while (column.next()) {
-          result.add(extractJdbcColumnFromResultSet(column));
-        }
-        jdbcColumns = result.toArray(new JdbcColumn[0]);
+      JdbcTable.Builder jdbcTableBuilder = getBasicJdbcTableInfo(table);
+
+      // 2.Get column information
+      List<JdbcColumn> jdbcColumns = new ArrayList<>();
+      ResultSet columns = getColumns(connection, databaseName, tableName);
+      while (columns.next()) {
+        JdbcColumn.Builder columnBuilder = getBasicJdbcColumnInfo(columns);
+        boolean autoIncrement = getAutoIncrementInfo(columns);
+        columnBuilder.withAutoIncrement(autoIncrement);
+        jdbcColumns.add(columnBuilder.build());
       }
-      return new JdbcTable.Builder()
-          .withName(tableName)
-          .withColumns(jdbcColumns)
-          .withComment(comment)
-          .withProperties(properties)
-          .withAuditInfo(AuditInfo.EMPTY)
-          .build();
-    } catch (final SQLException se) {
-      throw this.exceptionMapper.toGravitinoException(se);
+      jdbcTableBuilder.withColumns(jdbcColumns.toArray(new JdbcColumn[0]));
+
+      // 3.Get index information
+      List<Index> indexes = getIndexes(databaseName, tableName, connection.getMetaData());
+      jdbcTableBuilder.withIndexes(indexes.toArray(new Index[0]));
+
+      // 4.Get table properties
+      Map<String, String> tableProperties = getProperties(connection, tableName);
+      jdbcTableBuilder.withProperties(tableProperties);
+
+      // 5.Leave the information to the bottom layer to append the table
+      correctJdbcTableFields(connection, tableName, jdbcTableBuilder);
+      return jdbcTableBuilder.build();
+    } catch (SQLException e) {
+      throw exceptionMapper.toGravitinoException(e);
     }
   }
 
+  protected Map<String, String> getProperties(Connection connection, String tableName)
```

Comment:
rename to getTableProperties?   
please add description about will add extra properties like comment and will be filtered in xx